### PR TITLE
feat(forms+components): Support labelProps from UI schema

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
With #3378, it's possible to pass down `labelProps` to `FieldTemplate` that will be forwarded in `<label />`.
However, parent needs to pass it to `FieldTemplate`.

**What is the chosen solution to this problem?**
Forward prop from schema in all fields widgets where a label is shown.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Actions/Action/Action.component.js [hunk 1]
```diff
@@ -58,11 +58,7 @@ function getActionComponent({ displayMode, getComponent }) {
  * @param {ActionProps}
  */
 function Action({ displayMode, getComponent, ...props }) {
-	const ActionComponent = getActionComponent({
-		displayMode,
-		getComponent,
-		...props,
-	});
+	const ActionComponent = getActionComponent({ displayMode, getComponent });
```
Comment (sgangnant) [inline packages/components/src/Actions/Action/Action.component.js:61]: No need to spread `props`, `getActionComponent` only reads `displayMode` and `getComponent`

File: packages/components/src/ListView/Header/Header.component.js [hunk 2]
```diff
@@ -28,7 +27,7 @@ function getAction(action, index) {
 			icon={action.icon}
 			onClick={onClick}
 			disabled={action.disabled}
-			btooltipPlacement="bottom"
+			tooltipPlacement="bottom"
```
Comment (sgangnant) [inline packages/components/src/ListView/Header/Header.component.js:30]: Fixed typo

File: packages/components/src/ListView/Header/Header.component.js [hunk 3]
```diff
@@ -8,8 +9,6 @@ import getDefaultT from '../../translate';
 import Action from '../../Actions/Action';
 import theme from './Header.scss';
 
-import headerPropTypes from './Header.propTypes';
```
Comment (sgangnant) [inline packages/components/src/ListView/Header/Header.component.js:11]: PropTypes were written in a separate file, only used by this component 🤷

File: packages/components/src/ListView/Header/Header.component.js [hunk 4]
```diff
@@ -81,7 +80,15 @@ function Header({ headerDefault, headerLabel, nbItemsSelected, nbItems, required
...
-Header.propTypes = headerPropTypes;
+Header.propTypes = {
+	headerDefault: PropTypes.arrayOf(PropTypes.shape(Action.propTypes)).isRequired,
+	headerLabel: PropTypes.string,
+	labelProps: PropTypes.object,
+	required: PropTypes.bool,
+	nbItems: PropTypes.number,
+	nbItemsSelected: PropTypes.number,
+	t: PropTypes.func,
+};
```
Comment (sgangnant) [inline packages/components/src/ListView/Header/Header.component.js:91]: Copy/paste from deleted file

File: packages/forms/src/UIForm/fields/CheckBox/displayMode/TextMode.component.js [hunk 5]
```diff
@@ -8,7 +8,12 @@ import theme from './TextMode.scss';
 
 export default function TextModeCheckBox({ id, schema, value }) {
 	return (
-		<FieldTemplate id={id} label={schema.title || value} className={theme.checkbox}>
+		<FieldTemplate
+			id={id}
+			label={schema.title || value}
+			labelProps={schema.labelProps}
```
Comment (sgangnant) [inline packages/forms/src/UIForm/fields/CheckBox/displayMode/TextMode.component.js:14]: Only `labelProps` is new

File: packages/forms/src/UIForm/fields/Text/Text.component.test.js [hunk 6]
```diff
@@ -16,161 +16,130 @@ describe('Text field', () => {
...
 
+	const defaultProps = {
+		id: 'myForm',
+		isValid: true,
+		errorMessage: 'My error message',
+		onChange: jest.fn(),
+		onFinish: jest.fn(),
+		value: 'toto',
+		schema: defaultSchema,
+	};
```
Comment (sgangnant) [inline packages/forms/src/UIForm/fields/Text/Text.component.test.js:27]: Factorized test props

**Review by ncomont — APPROVED** [review]

**Comment by vbocquet** [comment]
Read the code, seems good. Just trying to reproduce in the UI Form playground here: http://3379.talend.surge.sh/forms/?path=/story/uiform-v2-playground--form-tester

But i can't be able to find a way to have my props on the label. Here is the json i build and paste:

```json
{
  "doc": "Text/README.md",
  "jsonSchema": {
    "type": "object",
    "title": "Text",
    "properties": {
      "text": {
        "type": "string"
      },
      "number": {
        "type": "number"
      },
      "numberMinMax": {
        "type": "number",
        "minimum": 0,
        "maximum": 7
      },
      "numberMinMaxStep": {
        "type": "number",
        "minimum": 1.5,
        "maximum": 3,
        "step": 0.1
      },
      "password": {
        "type": "string"
      }
    }
  },
  "uiSchema": [
    {
      "key": "text",
      "title": "Simple text",
      "data-test": "simple.text"
    },
    {
      "key": "number",
      "title": "Number",
      "labelProps": {
		"data-feature": "hello"      	
      }
    },
    {
      "key": "numberMinMax",
      "title": "Number (with min and max)"
    },
    {
      "key": "numberMinMaxStep",
      "title": "Number (with min, max and step)"
    },
    {
      "key": "password",
      "title": "Password",
      "type": "password"
    }
  ],
  "properties": {}
}
```

**Review by vbocquet — APPROVED** [review]

**Pull request merged**